radixdlt/src/main/java/com/radixdlt/consensus/ValidatingEventCoordinator.java

```diff
@@ -113,7 +117,7 @@ private void sync(QuorumCertificate qc, ECPublicKey node) throws SyncException {
 					this.getShortName(), this.getShortName(node.euid()), vertexId.toString().substring(0, 6));
 				return networkSender.getVertex(vertexId, node)
 					.doOnSuccess(v -> log.info("{}: Received GET_VERTEX Response: {}", this.getShortName(), v))
-					.takeUntil(this.pacemaker.nextLocalTimeout());
+					.takeUntil(this.pacemakerRx.timeout(this.pacemaker.getCurrentView()));
```

Comment:
Is there a possible race condition between the call to `timeout` and `getCurrentView` here?

Reply:
nope, `.timeout` works whether the timeout already occurred or not via use of `BehaviorSubject` in ScheduledTimeoutSender. Rx magic!